xarray/test/test_dataarray.py

```diff
@@ -2268,3 +2268,35 @@ def test_dot(self):
             da.dot(dm.values)
         with self.assertRaisesRegexp(ValueError, 'no shared dimensions'):
             da.dot(DataArray(1))
+
+    def test_binary_op_join_setting(self):
+        """
+        A test method to verify the ability to set binary operation join kwarg
+        ("inner", "outer", "left", "right") via xr.set_options().
+        """
+        # First we set up a data array
+        xdim, ydim, zdim = 'x', 'y', 'z'
+        xcoords, ycoords, zcoords = ['a', 'b', 'c'], [-2, 0, 2], [0, 1, 2]
+        total_size = len(xcoords) * len(ycoords) * len(zcoords)
+        # create a 3-by-3-by-3 data array
+        arr = xr.DataArray(np.arange(total_size).\
+                           reshape(len(xcoords),len(ycoords),len(zcoords)),
+                           [(xdim, xcoords), (ydim, ycoords),(zdim, zcoords)])
+        # now create a data array with the last x slice missing
+        missing_last_x = arr[0:-1,:,:].copy()
+        # create another data array with the last z slice missing
+        missing_last_z = arr[:,:,0:-1].copy()
+        # because the default in OPTIONS is join="inner", we test "outer" first
+        xr.set_options(join="outer")
+        result = missing_last_x + missing_last_z
+        self.assertTrue(result.shape == arr.shape)
```

Comment:
Instead of checking specific properties, construct the full expected result (e.g., by calling `align` with `join='outer'` on the inputs) and use `assertDataArrayEqual` to verify that expect result is the result you get.